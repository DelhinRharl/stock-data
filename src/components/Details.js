import React from 'react';

const Details = (item) => {
  const {
    symbol, fillingDate, date, period, netReceivables, cashAndCashEquivalents,
  } = item;
  return (
    <div>
      <div className="bg-gradient-to-r from-[#3dc2ff] to-[#0058bd] grid h-min-48 p-2 border-b-2 border-cyan-500">
        <div>
          <span className="text-black font-extrabold text-xl ">{symbol}</span>
        </div>
        <span className="text-blue-900">
          Period:
          <span className="font-semibold">{period}</span>
        </span>
        <span className="text-blue-900">
          {' '}
          Filling Date:
          <span className="font-semibold text-black">{fillingDate}</span>
        </span>
        <span className="text-black">
          {' '}
          Date:
          <span className="font-semibold">
            {' '}
            {date}
          </span>
        </span>
        <span className="text-black">
          {' '}
          Receivables:
          <span className="text-green-900 font-bold">{netReceivables}</span>
        </span>
        <span className="text-black">
          {' '}
          cashAndCashEquivalents:
          <span className="text-green-900 font-bold">{cashAndCashEquivalents}</span>
        </span>
      </div>
    </div>
  );
};

export default Details;
