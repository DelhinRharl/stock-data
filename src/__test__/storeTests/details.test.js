import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Details from '../../components/Details';

describe('Main data  tests', () => {
  test('should render the data', () => {
    const data = {
      ticker: 'F',
      changes: 0.360001,
      price: '20.66',
      changesPercentage: '1.7734021',
      companyName: 'Ford Motor Company',
    };
    const details = render(
      <BrowserRouter>
        <Details data={data} />
      </BrowserRouter>,
    );

    expect(details).toMatchSnapshot();
  });
});
