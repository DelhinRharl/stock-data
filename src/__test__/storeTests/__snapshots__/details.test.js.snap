// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main data  tests should render the data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="[#0e128a] bg-gradient-to-b from-[#0e128a] to-[#01034d] grid h-min-48 p-2"
        >
          <div>
            <span
              class="text-[#cc2263] font-bold font-cer"
            />
          </div>
          <span
            class="text-gray-400"
          >
            Period:
          </span>
          <span
            class="text-blue-300"
          >
             Filling Date:
          </span>
          <span
            class="text-white"
          >
             
            Date:
          </span>
          <span
            class="text-green-400 "
          >
             Receivables: 
            <span
              class="text-[#cc2263]"
            />
          </span>
          <span
            class="text-green-400 "
          >
             cashAndCashEquivalents: 
            <span
              class="text-[#cc2263]"
            />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="[#0e128a] bg-gradient-to-b from-[#0e128a] to-[#01034d] grid h-min-48 p-2"
      >
        <div>
          <span
            class="text-[#cc2263] font-bold font-cer"
          />
        </div>
        <span
          class="text-gray-400"
        >
          Period:
        </span>
        <span
          class="text-blue-300"
        >
           Filling Date:
        </span>
        <span
          class="text-white"
        >
           
          Date:
        </span>
        <span
          class="text-green-400 "
        >
           Receivables: 
          <span
            class="text-[#cc2263]"
          />
        </span>
        <span
          class="text-green-400 "
        >
           cashAndCashEquivalents: 
          <span
            class="text-[#cc2263]"
          />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
